statsmodels/tsa/arima/estimators/hannan_rissanen.py

```diff
@@ -106,23 +114,46 @@ def hannan_rissanen(endog, ar_order=0, ma_order=0, demean=True,
     # Create a spec, just to validate the initial autoregressive order
     _ = SARIMAXSpecification(endog, ar_order=initial_ar_order)
 
+    # Unpack fixed and free ar/ma lags, ix, and params (fixed only)
+    params_info = _package_fixed_and_free_params_info(
+        fixed_params, spec.ar_lags, spec.ma_lags
+    )
+
     # Compute lagged endog
-    # (`ar_ix`, and `ma_ix` below, are to account for non-consecutive lags;
-    # for indexing purposes, must have dtype int)
-    ar_ix = np.array(spec.ar_lags, dtype=int) - 1
-    lagged_endog = lagmat(endog, max_ar_order, trim='both')[:, ar_ix]
+    lagged_endog = lagmat(endog, max_ar_order, trim='both')
 
     # If no AR or MA components, this is just a variance computation
     if max_ma_order == 0 and max_ar_order == 0:
         p.sigma2 = np.var(endog, ddof=0)
         resid = endog.copy()
     # If no MA component, this is just CSS
     elif max_ma_order == 0:
-        mod = OLS(endog[max_ar_order:], lagged_endog)
-        res = mod.fit()
-        resid = res.resid
-        p.ar_params = res.params
-        p.sigma2 = res.scale
+        # extract 1) lagged_endog with free params; 2) lagged_endog with fixed
+        # params; 3) endog residual after applying fixed params if applicable
+        X_with_free_params = lagged_endog[:, params_info.free_ar_ix]
+        X_with_fixed_params = lagged_endog[:, params_info.fixed_ar_ix]
+        y = endog[max_ar_order:]
+        if X_with_fixed_params.shape[1] != 0:
+            y -= X_with_fixed_params.dot(params_info.fixed_ar_params)
+
+        # no free ar params -> variance computation on the endog residual
+        if X_with_free_params.shape[1] == 0:
+            p.ar_params = params_info.fixed_ar_params
+            p.sigma2 = np.var(y, ddof=0)
+            resid = y.copy()
+        # otherwise OLS with endog residual (after applying fixed params) as y,
+        # and lagged_endog with free params as X
+        else:
+            mod = OLS(y, X_with_free_params)
+            res = mod.fit()
+            resid = res.resid
+            p.sigma2 = res.scale
+            p.ar_params = _stitch_fixed_and_free_params(
```

Comment:
Using a simple `_stitch_fixed_and_free_params` function here to 1) combine free and fixed params, and 2) sort combined params by the order of lag. Please see `_stitch_fixed_and_free_params` below.